src/configuration.ts

```diff
@@ -90,7 +90,8 @@ const BUILT_IN_CONFIG = /^tslint:(.*)$/;
  * of the search for a configuration.
  * @returns Load status for a TSLint configuration object
  */
-export function findConfiguration(configFile: string | null, inputFilePath: string): IConfigurationLoadResult {
+export function findConfiguration(configFile: string | null, inputFilePath: string): IConfigurationLoadResult;
+export function findConfiguration(configFile: string, inputFilePath?: string): IConfigurationLoadResult {
     const configPath = findConfigurationPath(configFile, inputFilePath);
```

Comment:
you will get a compile error on this line, because the parameter types do not match any overload.
You could work around it with this dirty hack:
```ts
findConfigurationPath(configFile, inputFilePath!); // note the ! after inputFilePath
```
Otherwise you need an if..else and still need to use a nonNullAssertion